### ISSUE TITLE
mAP does not show up on tensorboard for Object detection API

### ISSUE THREAD

**Issue by Shameendra** [tensorflow/models#5882]
Hi i am using the Google Object Detection API to train on my own data. However, when i run the eval.py(from legacy folder) in order to see evaluation results and then run tensorboard, just images and graphs show up but no scalars like mAP. the command i am using is

`python eval.py --logtostderr --checkpoint_dir=training/ --eval_dir=evaluation/ --pipeline_config_path=training/faster_rcnn_inception_v2_pedestrians.config`

Can anyone please tell me how do i resolve this? Do i need to add some extra lines in eval.py or is there some other things that i should do?

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
What is the top-level directory of the model you are using
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by Ekinkit** [comment]
Hello,I have the same question. Did you find the solution to deal with that? I can only find the loss in my tensorboard while in my final project I need to give the mAP.

**Comment by Shameendra** [comment]
Hi yes the solution was to simply wait. Eval.py can take several minutes depending on your computer. For me it took about 30 min before the mAP showed up on TensorBoard.  Remember i used eval.py from legacy and not model_main,py

**Comment by Ekinkit** [comment]
> Hi yes the solution was to simply wait. Eval.py can take several minutes depending on your computer. For me it took about 30 min before the mAP showed up on TensorBoard. Remember i used eval.py from legacy and not model_main,py

hello~I have used eval.py to evaluate my model. But what appear in the tensorboard are only the images predicted. I don't see the mAP. Do I need to add some code to show the mAP?

**Comment by snoopypi** [comment]
> > Hi yes the solution was to simply wait. Eval.py can take several minutes depending on your computer. For me it took about 30 min before the mAP showed up on TensorBoard. Remember i used eval.py from legacy and not model_main,py
> 
> hello~I have used eval.py to evaluate my model. But what appear in the tensorboard are only the images predicted. I don't see the mAP. Do I need to add some code to show the mAP?

Did you find a solution? Thank you*

**Comment by Shameendra** [comment]
Hi, i know this is very latre but for me what worked was waiting for checkpoints to be saved. i did not add any code to see tensorboard on eval.py

**Comment by snoopypi** [comment]
Hi ok thank you and what train file did you use? mAP showed suddenly up in tensorboard?

**Comment by Shameendra** [comment]
I used the train.py in legacy folder

**Comment by snoopypi** [comment]
> I used the train.py in legacy folder. And yes mAP showed up after checkpoints were created

Ok thank you I will try it again!!

**Comment by snoopypi** [comment]
> I used the train.py in legacy folder. And yes mAP showed up after checkpoints were created

Hey sorry just one more question. For the eval.py. Did you use path to your .config file from training? What is written there in the eval_configs?

Thank you!

**Comment by kaisubr** [comment]
@snoopypi Yes, the config file should be the same. I don't think the `eval_config_path` flag is necessary. The command might look like this:
```
!python "path/to/your/eval.py" --logtostderr  --pipeline_config_path="training/faster_rcnn_inception_v2_coco.config" --checkpoint_dir="training/log" --eval_dir="training/log/eval_py"
```

However, you might want to look into `model_main.py` which performs evaluations between training steps.
